classes/patterns/class-issues.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * Issues class.
+ *
+ * @package P4GBKS
+ * @since 0.1
+ */
+
+namespace P4GBKS\Patterns;
+
+/**
+ * Issues pattern includes:
+ * Column, Image, Heading, Paragraph, Media & Text.
+ *
+ * @package P4GBKS\Patterns
+ */
+class Issues extends Block_Pattern {
+
+	/**
+	 * Returns the pattern name.
+	 */
+	public static function get_name(): string {
+		return 'p4/issues';
+	}
+
+	/**
+	 * Returns the template for one column.
+	 */
+	public static function get_column_template(): string {
+		return '
+			<!-- wp:column {"verticalAlignment":"center","width":"25%","backgroundColor":"white"} -->
+			<div class="wp-block-column is-vertically-aligned-center has-white-background-color has-background" style="flex-basis:25%">
+				<!-- wp:group {"layout":{"wideSize":"80%","contentSize":"100%"}} -->
+				<div class="wp-block-group"><!-- wp:spacer {"height":"16px"} -->
+					<div style="height:16px" aria-hidden="true" class="wp-block-spacer"></div>
+					<!-- /wp:spacer -->
+					<!-- wp:media-text {"mediaWidth":20,"mediaSizeSlug":"thumbnail","isStackedOnMobile":false,"verticalAlignment":"center","imageFill":false} -->
+					<div class="wp-block-media-text alignwide is-vertically-aligned-center" style="grid-template-columns:20% auto"><figure class="wp-block-media-text__media"></figure>
+					<div class="wp-block-media-text__content">
+					<!-- wp:paragraph {"align":"left","placeholder":"' . __( 'Enter text', 'planet4-blocks-backend' ) . '","style":{"typography":{"fontSize":"1rem","fontStyle":"normal","fontWeight":"700"}}} -->
+					<p class="has-text-align-left" style="font-size:1rem;font-style:normal;font-weight:700"></p>
+					<!-- /wp:paragraph --></div></div>
+					<!-- /wp:media-text -->
+					<!-- wp:spacer {"height":"16px"} -->
+					<div style="height:16px" aria-hidden="true" class="wp-block-spacer"></div>
+					<!-- /wp:spacer --></div>
+				<!-- /wp:group --></div>
+			<!-- /wp:column -->
+		';
+	}
+
+	/**
+	 * Returns the pattern config.
+	 */
+	public static function get_config(): array {
+		return [
+			'title'      => __( 'Issues', 'planet4-blocks-backend' ),
+			'categories' => [ 'planet4' ],
+			'content'    => '
+				<!-- wp:group {"backgroundColor":"grey-10","className":"alignfull"} -->
```

Comment:
Can we do this without needing 2 groups? The block's complexity would be already a bit more manageable with 1 level less.

Reply:
The first group is the wide block that needs to be fit to the screen's width. And the second group is `container` class.

Reply:
Yeah this one makes sense and doesn't complicated it that much.

It's mostly a problem in the row of items below, as there the complexity is repeated for every element. That makes operations like selecting the right block in the editor very annoying. The paragraphs are currently 8 levels deep. While for a user it seems like the block would only need 3 or 4.
![Screenshot from 2022-04-19 22-02-00](https://user-images.githubusercontent.com/7604138/164088441-da44f570-1feb-4133-9340-5499fc20c954.png)

Reply:
I created the [same block using the "Row" block](https://www-dev.greenpeace.org/gutenberg/wp-admin/post.php?post=27114&action=edit) (which is enabled because on the gutenberg instance we don't severely limit the core block functionality).
With just 1 additional CSS rule, it looks identical to what's currently in the PR.
![Screenshot from 2022-04-19 22-02-29](https://user-images.githubusercontent.com/7604138/164088739-b2e255f8-073d-4482-b3e1-c6f11e9cb5ee.png)

Reply:
I completely agree to use `Row` instead of `Column`, in fact that was my initial plan. But the problem was that we needed -at least- a bit customization and it wasn't what we wanted to achieve. 

If we're fine to go with a bit customization then I suggest to use this [deleted file](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/811/commits/a917eaf2373318fa8cacc1cc27aac36a7c4696b6#diff-ac498fa0f16734d2b3a1af452585c6f5828b5612676c2e2b6c6dddc0409781b5) (with a couple of changes) and then provide to the NRO, the CSS in case they need it just by adding the classname through the editor panel (additional settings). 

What do you think?

Reply:
> But the problem was that we needed -at least- a bit customization and it wasn't what we wanted to achieve.

Can you elaborate? Which customization, and why "wasn't it what we wanted to achieve"?

Reply:
> If we're fine to go with a bit customization then I suggest to use this deleted file (with a couple of changes) and then provide to the NRO, the CSS in case they need it just by adding the classname through the editor panel

In my example you see I only had to add 1 CSS rule to set a out padding on the media and text block. That seems a pretty common use case, so we could register it as a block style. Then we could use it every time instead of nesting blocks to achieve padding.

Reply:
> Can you elaborate? Which customization, and why "wasn't it what we wanted to achieve"?
For example `box-shadow`, `border-radius`, `font-family`, `margin` between image and text. In fact, I had to add those styles to [this GPBR page](https://www-dev.greenpeace.org/brasil/trabalhe-conosco-2/) because they requested that.

Reply:
> With just 1 additional CSS rule, it looks identical to what's currently in the PR.

Wouldn't that require a css class explicitly for this pattern?

Reply:
No, we could register the style on the Media and Text block type, it currently doesn't have any block styles. It's often the case when you use this block that it needs padding. For now we had to add additional blocks just to get this padding right. With a style it's just 1 click in the sidebar. See also [this message](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/811#discussion_r854049107).

Reply:
I really like to register styles, but I think that the padding classes [you added here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/832) are very specific to this block. What do you think?

Reply:
You mean this style is not reusable in other cases? :thinking:

Reply:
You can register the same style for multiple blocks. I added that to the PR now, should be on the test instance soon.

Reply:
> You mean this style is not reusable in other cases?

Yes, but if you want to apply more padding to another `media-text` you should override it.

Reply:
Ah you mean the exact padding values. Well we do have a sort of design system, so we can create some values that are likely to work in a lot of cases.

For now I kept it simple in the PR and set a single padding of 8 / 16 / 32 px. I also included `align-self: stretch` as it seemed safe to do. How do you suggest making this more generic?

Reply:
Let's continue the padding conversation on this [PR](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/832/files)